### PR TITLE
[WW-4889] Adds XML handler based on Apache Juneau

### PR DESCRIPTION
**Pull request by lukaszlenart**
Implements [WW-4889](https://issues.apache.org/jira/browse/WW-4889)

### CODE REVIEW

File: plugins/rest/src/test/java/org/apache/struts2/rest/handler/JuneauXmlHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        // given
+        SimpleBean obj = new SimpleBean();
+
+        // when
+        Reader in = new StringReader(xml);
+        handler.toObject(ai, in, obj);
+
+        // then
+        assertNotNull(obj);
+        assertEquals(obj.getName(), "Jan");
```
Comment (sepe81) [inline plugins/rest/src/test/java/org/apache/struts2/rest/handler/JuneauXmlHandlerTest.java]: The params order for `assertEquals` should be expected, actual.

**Comment by lukaszlenart** [comment]
Just an alternative, I rather thought about switching to JacksonXML as a default to handle XMLs - it's already there.

**Comment by yasserzamani** [comment]
A test is failing. It seems generated xml elements order is different than your local one pc.

**Comment by lukaszlenart** [comment]
@yasserzamani thanks, fixed :)

**Comment by yasserzamani** [comment]
LGTM 👍

**Pull request merged**